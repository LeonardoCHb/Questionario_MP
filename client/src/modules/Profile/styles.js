import { makeStyles } from "@material-ui/core/styles";

export default makeStyles((theme) => ({
  ProfileContainer: {
    display: "flex",
  },
  AvatarContainer: {
    marginTop: theme.spacing(3),
    marginRight: theme.spacing(3),
    width: "auto",
    padding: 0,
  },
  AvatarPaper: {
    marginBottom: theme.spacing(2),
    padding: theme.spacing(2),
    width: 250,
    display: "flex",
  },
  AvatarImage: {
    width: theme.spacing(15),
    marginBottom: theme.spacing(3),
    height: theme.spacing(15),
  },
  AvatarContainerInfo: {
    flex: 1,
    flexDirection: "column",
    justifyContent: "flex-end",
  },
  QuizPaper: {
    marginBottom: theme.spacing(4),
    padding: theme.spacing(2),
    flex: 1,
    alignItems: "center",
  },
  QuizContainer: {
    flex: 1,
    marginTop: theme.spacing(3),
    padding: 0,
  },
}));
