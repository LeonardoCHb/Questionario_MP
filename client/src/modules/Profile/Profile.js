// estilização
import {
  Avatar,
  Box,
  Paper,
  CardActions,
  Container,
  Tab,
  Tabs,
  Typography,
} from "@material-ui/core";
import PropTypes from "prop-types";
// react e redux
import React, { useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";

// components
import { findAllUserQuizzes, findAllUserResponses } from "../../actions/quiz";
import UserQuizzes from "./components/Quizzes/UserQuizzes";
import UserResponses from "./components/Responses/UserResponses";
import styles from "./styles.js";

// INICIO DAS FUNCOES PARA FUNCIONAMENTO DO COMPONENTE
function TabPanel(props) {
  const { children, value, index, ...other } = props;

  return (
    <div
      role="tabpanel"
      hidden={value !== index}
      id={`nav-tabpanel-${index}`}
      aria-labelledby={`nav-tab-${index}`}
      {...other}
    >
      {value === index && (
        <Box p={3}>
          <Typography>{children}</Typography>
        </Box>
      )}
    </div>
  );
}

TabPanel.propTypes = {
  children: PropTypes.node,
  index: PropTypes.any.isRequired,
  value: PropTypes.any.isRequired,
};

function a11yProps(index) {
  return {
    id: `nav-tab-${index}`,
    "aria-controls": `nav-tabpanel-${index}`,
  };
}
function LinkTab(props) {
  return (
    <Tab
      component="a"
      onClick={(event) => {
        event.preventDefault();
      }}
      {...props}
    />
  );
}
// FIM DAS FUNCOES AUXILIARES

const useStyles = styles;

// INICIO DO COMPONENTE
const Profile = () => {
  const classes = useStyles();
  const [user] = useState(JSON.parse(localStorage.getItem("profile")));
  const [value, setValue] = React.useState(0);
  const handleChange = (event, newValue) => setValue(newValue);
  const dispatch = useDispatch();
  const quizzes = useSelector((state) => state.quizzesUser);
  const responses = useSelector((state) => state.response);

  useEffect(() => {
    dispatch(findAllUserQuizzes());
    dispatch(findAllUserResponses());
  }, []);

  useEffect(() => {
    dispatch(findAllUserResponses());
  }, [quizzes]);

  return (
    <>
      <Container className={classes.ProfileContainer}>
        {/* Container do Avatar */}
        <Container className={classes.AvatarContainer}>
          <Paper className={classes.AvatarPaper}>
            <CardActions className={classes.AvatarContainerInfo}>
              <Avatar
                className={classes.AvatarImage}
                alt={user?.result.name ? user?.result.name : "My robohash"}
                variant="circular"
                src={
                  user?.result.imageUrl
                    ? user?.result.imageUrl
                    : `https://robohash.org/${user?.result._id}`
                }
              />
              <Typography variant="h5">{user.result.name}</Typography>
              <Typography variant="p">{user.result.email}</Typography>
            </CardActions>
          </Paper>
        </Container>
        {/* Container dos Quiz Criados/Respondidos */}
        <Container className={classes.QuizContainer}>
          <Paper className={classes.QuizPaper}>
            <Tabs
              centered
              value={value}
              onChange={handleChange}
              aria-label="nav tabs example"
            >
              <LinkTab label="Questionarios Criados" {...a11yProps(0)} />
              <LinkTab label="Questionarios Respondidos" {...a11yProps(1)} />
            </Tabs>
            <TabPanel value={value} index={0}>
              <Typography variant="h5" align="justify">
                Questionarios criados: {quizzes ? quizzes.length : 0}
              </Typography>
              <UserQuizzes quizzes={quizzes} />
            </TabPanel>
            <TabPanel value={value} index={1}>
              <Typography variant="h5" align="justify">
                Nº de questionarios respondidos:{" "}
                {responses ? responses.length : 0}
              </Typography>
              <UserResponses responses={responses} />
            </TabPanel>
          </Paper>
        </Container>
      </Container>
    </>
  );
};

export default {
  routeProps: {
    path: "/profile",
    component: Profile,
  },
  name: "Profile",
};
