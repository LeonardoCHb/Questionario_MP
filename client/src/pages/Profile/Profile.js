import { Paper } from "@material-ui/core";
import Avatar from "@material-ui/core/Avatar";
import Box from "@material-ui/core/Box";
import CardActions from "@material-ui/core/CardActions";
import Container from "@material-ui/core/Container";
import Tab from "@material-ui/core/Tab";
import Tabs from "@material-ui/core/Tabs";
import Typography from "@material-ui/core/Typography";
import PropTypes from "prop-types";
import React, { useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";

import { findAllUserQuizzes, findAllUserResponses } from "../../actions/quiz";
import Quiz from "../../components/CreatorQuizzes/CreatorQuizzes.js";
import Responses from "../../components/CreatorResponses/CreatorResponses.js";
import styles from "./styles.js";

// INICIO DAS FUNCOES PARA FUNCIONAMENTO DO COMPONENTE
function TabPanel(props) {
  const { children, value, index, ...other } = props;

  return (
    <div
      role="tabpanel"
      hidden={value !== index}
      id={`nav-tabpanel-${index}`}
      aria-labelledby={`nav-tab-${index}`}
      {...other}
    >
      {value === index && (
        <Box p={3}>
          <Typography>{children}</Typography>
        </Box>
      )}
    </div>
  );
}

TabPanel.propTypes = {
  children: PropTypes.node,
  index: PropTypes.any.isRequired,
  value: PropTypes.any.isRequired,
};

function a11yProps(index) {
  return {
    id: `nav-tab-${index}`,
    "aria-controls": `nav-tabpanel-${index}`,
  };
}
function LinkTab(props) {
  return (
    <Tab
      component="a"
      onClick={(event) => {
        event.preventDefault();
      }}
      {...props}
    />
  );
}
// FIM DAS FUNCOES AUXILIARES

const useStyles = styles;

// INICIO DO COMPONENTE
const Profile = () => {
  const classes = useStyles();
  const [user] = useState(JSON.parse(localStorage.getItem("profile")));
  const [value, setValue] = React.useState(0);
  const handleChange = (event, newValue) => setValue(newValue);
  const dispatch = useDispatch();
  const quizzes = useSelector((state) => state.quizzesUser);
  const responses = useSelector((state) => state.response);

  useEffect(() => {
    dispatch(findAllUserQuizzes());
    dispatch(findAllUserResponses());
  }, []);

  useEffect(() => {
    dispatch(findAllUserResponses());
  }, [quizzes]);

  return (
    <>
      <Container className={classes.ProfileContainer}>
        {/* Container do Avatar */}
        <Container className={classes.AvatarContainer}>
          <Paper className={classes.AvatarPaper}>
            <CardActions className={classes.AvatarContainerInfo}>
              <Avatar
                className={classes.AvatarImage}
                alt={user?.result.name ? user?.result.name : "My robohash"}
                variant="circular"
                src={
                  user?.result.imageUrl
                    ? user?.result.imageUrl
                    : `https://robohash.org/${user?.result._id}`
                }
              />
              <Typography variant="h5">{user.result.name}</Typography>
              <Typography variant="p">{user.result.email}</Typography>
            </CardActions>
          </Paper>
        </Container>
        {/* Container dos Quiz Criados/Respondidos */}
        <Container className={classes.QuizContainer}>
          <Paper className={classes.QuizPaper}>
            <Tabs
              centered
              value={value}
              onChange={handleChange}
              aria-label="nav tabs example"
            >
              <LinkTab label="Questionarios Criados" {...a11yProps(0)} />
              <LinkTab label="Questionarios Respondidos" {...a11yProps(1)} />
            </Tabs>
            <TabPanel value={value} index={0}>
              <Typography variant="h5" align="justify">
                Questionarios criados: {quizzes ? quizzes.length : 0}
              </Typography>
              <Quiz quizzes={quizzes} />
            </TabPanel>
            <TabPanel value={value} index={1}>
              <Typography variant="h5" align="justify">
                NÂº de questionarios respondidos:{" "}
                {responses ? responses.length : 0}
              </Typography>
              <Responses responses={responses} />
            </TabPanel>
          </Paper>
        </Container>
      </Container>
    </>
  );
};
export default Profile;
