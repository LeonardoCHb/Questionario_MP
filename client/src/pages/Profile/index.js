import { Paper } from "@material-ui/core";
import Box from "@material-ui/core/Box";
import Grid from "@material-ui/core/Grid";
import Tab from "@material-ui/core/Tab";
import Tabs from "@material-ui/core/Tabs";
import Typography from "@material-ui/core/Typography";
import PropTypes from "prop-types";
import React from "react";

import Quiz from "./quiz.js";
import styles from "./styles.js";
function TabPanel(props) {
  const { children, value, index, ...other } = props;

  return (
    <div
      role="tabpanel"
      hidden={value !== index}
      id={`nav-tabpanel-${index}`}
      aria-labelledby={`nav-tab-${index}`}
      {...other}
    >
      {value === index && (
        <Box p={3}>
          <Typography>{children}</Typography>
        </Box>
      )}
    </div>
  );
}

TabPanel.propTypes = {
  children: PropTypes.node,
  index: PropTypes.any.isRequired,
  value: PropTypes.any.isRequired,
};

const useStyles = styles;
const Profile = () => {
  const classes = useStyles();
  function a11yProps(index) {
    return {
      id: `nav-tab-${index}`,
      "aria-controls": `nav-tabpanel-${index}`,
    };
  }
  function LinkTab(props) {
    return (
      <Tab
        component="a"
        onClick={(event) => {
          event.preventDefault();
        }}
        {...props}
      />
    );
  }
  const [value, setValue] = React.useState(0);

  const handleChange = (event, newValue) => {
    setValue(newValue);
  };
  return (
    <>
      <Paper className={classes.paper}>
        <Typography variant="h5">Nome: jonisvaldo</Typography>
        <Typography variant="h6">
          Nº de questionarios criados: 9999999
        </Typography>
        <Typography variant="h6">
          Nº de questionarios respondidos: 666
        </Typography>
      </Paper>
      <Paper className={classes.quizzes}>
        <Tabs
          // variant="fullWidth"
          value={value}
          onChange={handleChange}
          aria-label="nav tabs example"
        >
          <LinkTab label="Questionarios Criados" {...a11yProps(0)} />
          <LinkTab label="Questionarios Respondidos" {...a11yProps(1)} />
        </Tabs>
        <TabPanel value={value} index={0}>
          <Grid container item xs={13} spacing={3}>
            <Quiz />
          </Grid>
        </TabPanel>
        <TabPanel value={value} index={1}>
          <Grid container item xs={13} spacing={3}>
            AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
          </Grid>
        </TabPanel>
      </Paper>
    </>
  );
};
export default Profile;
