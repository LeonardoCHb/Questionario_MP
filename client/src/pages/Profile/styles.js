import { makeStyles } from "@material-ui/core/styles";

export default makeStyles((theme) => ({
  paper: {
    marginTop: theme.spacing(2),
    marginBottom: theme.spacing(2),
    marginLeft: theme.spacing(60),
    marginRight: theme.spacing(60),
    display: "flex",
    flexDirection: "column",
    // alignItems: "center",
    height: "20vh",
    padding: theme.spacing(2),
    background: "linear-gradient(to bottom, #0099ff 5%, #99ccff 86%)",
  },
  quizzes: {
    height: "100vh",
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    marginTop: theme.spacing(2),
    marginBottom: theme.spacing(2),
    marginLeft: theme.spacing(60),
    marginRight: theme.spacing(60),
    background: "linear-gradient(to bottom, #66ccff 0%, #ffcccc 81%)",
  },
  quiz: {
    width: "100%",
    height: "15rem",
    border: "1.5px",
    borderRadius: "1.5rem",
    alignItems: "center",
    display: "flex",
  },
}));
