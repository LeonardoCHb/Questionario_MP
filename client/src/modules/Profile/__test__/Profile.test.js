import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";

import Profile from "../Profile";

const user = {
  createdAt: "2021-04-30T23:54:53.614Z",
  _id: "608cce84d339170cd8ec9894",
  name: "Fernando",
  email: "fernando@email",
  password: "$2a$12$CDydrU0e8OwZZZNg0pWrgeuRBHaDimQplLKxBwNmDPvlrDJ24J69y",
  __v: 0,
};

localStorage.setItem("profile", JSON.stringify({ result: { ...user } }));

const ProfileComponent = Profile.routeProps.component;

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const initialState = [];
const store = mockStore(initialState);

it("Profile Ã© renderizado ", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <ProfileComponent />
      </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
