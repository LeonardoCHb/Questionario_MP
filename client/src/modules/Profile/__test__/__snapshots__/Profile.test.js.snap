// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Ã© renderizado  1`] = `
<div
  className="MuiContainer-root makeStyles-ProfileContainer-1 MuiContainer-maxWidthLg"
>
  <div
    className="MuiContainer-root makeStyles-AvatarContainer-2 MuiContainer-maxWidthLg"
  >
    <div
      className="MuiPaper-root makeStyles-AvatarPaper-3 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiCardActions-root makeStyles-AvatarContainerInfo-5 MuiCardActions-spacing"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circular makeStyles-AvatarImage-4"
        >
          <img
            alt="Fernando"
            className="MuiAvatar-img"
            src="https://robohash.org/608cce84d339170cd8ec9894"
          />
        </div>
        <h5
          className="MuiTypography-root MuiTypography-h5"
        >
          Fernando
        </h5>
        <p
          className="MuiTypography-root MuiTypography-body2"
        >
          fernando@email
        </p>
      </div>
    </div>
  </div>
  <div
    className="MuiContainer-root makeStyles-QuizContainer-7 MuiContainer-maxWidthLg"
  >
    <div
      className="MuiPaper-root makeStyles-QuizPaper-6 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiTabs-root"
      >
        <div
          className="MuiTabs-scroller MuiTabs-fixed"
          onScroll={[Function]}
          style={
            Object {
              "marginBottom": null,
              "overflow": "hidden",
            }
          }
        >
          <div
            aria-label="nav tabs example"
            className="MuiTabs-flexContainer MuiTabs-centered"
            onKeyDown={[Function]}
            role="tablist"
          >
            <a
              aria-controls="nav-tabpanel-0"
              aria-disabled={false}
              aria-selected={true}
              className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected"
              id="nav-tab-0"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex={0}
            >
              <span
                className="MuiTab-wrapper"
              >
                Questionarios Criados
              </span>
              <span
                className="PrivateTabIndicator-root-8 PrivateTabIndicator-colorSecondary-10 MuiTabs-indicator"
                style={Object {}}
              />
            </a>
            <a
              aria-controls="nav-tabpanel-1"
              aria-disabled={false}
              aria-selected={false}
              className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
              id="nav-tab-1"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex={-1}
            >
              <span
                className="MuiTab-wrapper"
              >
                Questionarios Respondidos
              </span>
            </a>
          </div>
        </div>
      </div>
      <div
        aria-labelledby="nav-tab-0"
        hidden={false}
        id="nav-tabpanel-0"
        role="tabpanel"
      >
        <div
          className="MuiBox-root MuiBox-root-12"
        >
          <h5
            className="MuiTypography-root MuiTypography-h5 MuiTypography-alignJustify"
          >
            Questionarios criados: 
            0
          </h5>
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            :(
          </p>
        </div>
      </div>
      <div
        aria-labelledby="nav-tab-1"
        hidden={true}
        id="nav-tabpanel-1"
        role="tabpanel"
      />
    </div>
  </div>
</div>
`;
